src/WebInterface.hs

```diff
@@ -228,24 +229,23 @@ viewGroupedProjectQueues projects = do
 
 -- Renders the contents of a list item with a link for a pull request.
 viewPullRequest :: ProjectInfo -> PullRequestId -> PullRequest -> Html
-viewPullRequest info (PullRequestId n) pullRequest =
-  let
-    url = format "https://github.com/{}/{}/pull/{}"
-      (Project.owner info, Project.repository info, n)
-  in do
-    a ! href (toValue url) $ toHtml $ Project.title pullRequest
-    span ! class_ "prId" $ toHtml $ "#" <> (show n)
-
-    case integrationStatus pullRequest of
-      Integrated _ (BuildPending (Just ciUrl)) -> do
-        span "  | "
-        a ! href (toValue ciUrl) $ "View in CI"
-
-      Integrated _ (BuildFailed (Just ciUrl)) -> do
-        span "  | "
-        a ! href (toValue ciUrl) $ "View in CI"
-
-      _ -> pure ()
+viewPullRequest info pullRequestId pullRequest = do
+  a ! href (toValue $ pullRequestUrl info pullRequestId) $ toHtml $ Project.title pullRequest
+  span ! class_ "prId" $ toHtml $ prettyPullRequestId pullRequestId
+
+  case integrationStatus pullRequest of
+    Integrated sha buildStatus -> do
+      span "  | "
+      case buildStatus of
+        (BuildStarted ciUrl)       -> ciLink ciUrl "🟡"
+        (BuildFailed (Just ciUrl)) -> ciLink ciUrl "❌"
+        _                          -> pure ()
+      a ! href (toValue $ commitUrl info sha) $ toHtml $ prettySha sha
```

Comment:
I think I would change this, having the link to CI only be shown as an icon makes it easier to miss. If I didn't realise I would probably first click the SHA and possibly miss the fact the icon is a completely different link.

Reply:
Fair enough.  I'll include a "CI build" link besides the hash when it is available.